### ISSUE TITLE
Errors are not cleared on file save and sometimes fixed errors come back

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#2192]
## Environment data

-   Language Server version: v2021.12.2
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10.1

## Expected behaviour

Fixed errors disappear after file save and they don't come back after unrelated code changes.

## Actual behaviour

https://user-images.githubusercontent.com/12860284/147288170-96b616d2-e756-4e8f-8741-5aacca11f4d0.mp4

## Code Snippet / Additional information

```python
from itertools import accumulate


cashflows = [1, 2]
lst = list(accumulate(cashflows, lambda bal, pmt: bal * 2 + pmt))
```

1. Start with the above code snippet that should trigger 5 errors (with the `strict` type checking).
2. Add `initial=0` to the `accumulate`. This will remove only 1 error.
3. Add `pass` below, see no changes, still 4 errors.
4. Remove `pass`, see all 4 errors disappear. This should have happened in step 2.
5. Add `pass` again, see 4 errors reappear again.

**Comment by erictraut** [comment]
I think I found and fixed the underlying bug. This will be addressed in the next release.

**Comment by judej** [comment]
@erictraut, I am seeing the errors return after step #5 above. Wasn't that meant to be fixed?

**Comment by erictraut** [comment]
Just to confirm, are you using a private (pre-release) build of pylance that has all of the latest pyright changes incorporated? I'm not able to repro this with the latest pyright build.

**Comment by judej** [comment]
I think the latest Pyright changes are not yet in the build I am using... Will validate in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202210-5-january-2022)
